PropertyChangedAnalyzers/RoslynVersionSupportExtensions.cs

```diff
@@ -0,0 +1,61 @@
+﻿namespace PropertyChangedAnalyzers
+{
+    using System;
+    using System.Reflection;
+    using Microsoft.CodeAnalysis;
+
+    /// <summary>
+    /// Enables compilation against older versions of Roslyn while accessing features from newer versions.
+    /// </summary>
+    internal static class RoslynVersionSupportExtensions
+    {
+        private static readonly Lazy<Func<SemanticModel, int, int>?> GetNullableContext = new Lazy<Func<SemanticModel, int, int>?>(
+            () => typeof(SemanticModel).GetInstanceMethodDelegate<Func<SemanticModel, int, int>>("GetNullableContext"));
+
+        /// <summary>
+        /// Evaluates <c>semanticModel.GetNullableContext(position).AnnotationsEnabled()</c> if the loaded version of
+        /// Roslyn contains these APIs. Otherwise, returns <see langword="false"/> since the loaded version of Roslyn
+        /// does not support C# 8.
+        /// </summary>
+        internal static bool NullableAnnotationsEnabled(this SemanticModel semanticModel, int position)
+        {
+            if (GetNullableContext.Value is { } getNullableContext)
+            {
+                var nullableContext = getNullableContext.Invoke(semanticModel, position);
+
+                // Roslyn's public flags enum value definition
+                const int AnnotationsEnabled = 1 << 1;
+
+                return (nullableContext & AnnotationsEnabled) != 0;
+            }
+
+            return false;
+        }
```

Comment:
It's really not that hard to support multiple versions of Roslyn when the new APIs are single method calls.

Reply:
We can bump Roslyn version and create a branch for VS2017 also.

Chances are reflection will not work in VS with the current dependency resolution graph.

Reply:
Oh, I forgot about that. What are the steps I need to test this? F5 debug the VSIX?

Reply:
The vsix may work or not :)

Reply:
When you discovered the problem, was it via nuget package reference at the time?